util/otbn_build.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import logging as log
+import os
+import shlex
+import subprocess
+import sys
+from pathlib import Path
+from typing import List
+
+log.basicConfig(level=log.INFO, format="%(message)s")
+
+repo_top = (Path(os.path.dirname(__file__)) / '../').resolve()
+
+
+def run_cmd(cmd, **kwargs):
+    cmd_str = ' '.join([shlex.quote(str(a)) for a in cmd])
+    log.info(cmd_str)
+    subprocess.run(cmd, **kwargs)
+
+
+def call_otbn_as(src_file: Path, out_file: Path):
+    otbn_as_cmd = os.environ.get('OTBN_AS',
+                                 str(repo_top / 'hw/ip/otbn/util/otbn-as'))
+    run_cmd([otbn_as_cmd, '-o', out_file, src_file], check=True)
+
+
+def call_otbn_ld(src_files: List[Path], out_file: Path):
+    otbn_ld_cmd = os.environ.get('OTBN_LD',
+                                 str(repo_top / 'hw/ip/otbn/util/otbn-ld'))
+    run_cmd([otbn_ld_cmd, '-o', out_file] + src_files, check=True)
+
+
+def call_rv32_objcopy(args: List[str]):
+    rv32_tool_objcopy = os.environ.get('RV32_TOOL_OBJCOPY',
+                                       'riscv32-unknown-elf-objcopy')
+    run_cmd([rv32_tool_objcopy] + args, check=True)
```

Comment:
Is there a reason these aren't passed as command-line arguments? If there is, I think it should be documented.

Reply:
I don't think there was an answer to this question?

Reply:
There is now a big block of documentation at the beginning of the file (and in the help output). The environment variables are used by this tool and by underlying tools, passing them in as environment variables is just simpler than passing them in as argument and then converting them into an environment variable within this script.